src/session_impl.cpp

```diff
@@ -1856,6 +1864,14 @@ namespace {
 		}
 #endif
 
+		// if no listen interfaces are specified, create sockets to use
+		// any interface
+		if (eps.empty())
+		{
+			eps.emplace_back(address_v4(), 0, "", transport::plaintext
+				, duplex::only_outgoing);
```

Comment:
There should be a socket for IPv6 as well, otherwise we can't announce to IPv6 only trackers. Binding to `::` is fine in this case.